### ISSUE TITLE
gradients not updating when I use norm?

### ISSUE THREAD

**Issue by demiguo** [pytorch/pytorch#3029]
When I include this in loss function, I got 50% accuracy and sometimes my model converge. (I'm guessing the gradients are not calculated properly).

loss2 = (self.l * torch.norm(self.linear_layer.weight, p=1))

However, If I change it to L1Loss - it works! I got a normal accuracy (89%):
loss2 = torch.nn.L1Loss()(self.linear_layer.weight, Variable(torch.zeros((self.vocab_size, 1))))

Why is that?
Thanks!

**Comment by apaszke** [comment]
These two examples are not necessarily equivalent, because you have that additonal `* self.l` in the first one. Maybe that's what's breaking your training

**Comment by apaszke** [comment]
Also, L1Loss contains a normalization constant (` / self.linear_layer.weight.numel()`), which `norm` doesn't have (it's just a sum)
